src/neo/Cryptography/MPT/BranchNode.cs

```diff
@@ -0,0 +1,50 @@
+using Neo.IO.Json;
+using System.IO;
+
+namespace Neo.Cryptography.MPT
+{
+    public class BranchNode : MPTNode
+    {
+        public const int ChildCount = 17;
```

Comment:
Where is this `17` from?

Reply:
Count of 4 bit max value from `0x00` to `0x0F`. And the empty scenario.

When the key is `0xab12`, we split it into `0x0a0b0102` as path. So there will be only 16 branches at most.

The 17th is for the scenario that there is different path length with one exactly point to the `BranchNode`, For example, if we have stored `0x0a0b0100` and `0x0a0b0a00`,  the 17th child is for `0x0a0b`
![image](https://user-images.githubusercontent.com/22324403/82187391-eb103d00-991e-11ea-8403-fc1d61f14189.png)